src/core/server/saved_objects/saved_objects_type_registry.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { deepFreeze } from '../../utils';
+import { SavedObjectsType } from './types';
+
+/**
+ * @internal
+ */
+export class SavedObjectTypeRegistry {
+  private types = new Map<string, SavedObjectsType>();
+
+  public registerType(type: SavedObjectsType) {
+    if (this.types.has(type.name)) {
+      throw new Error(`Type '${type.name}' is already registered`);
+    }
+    this.types.set(type.name, deepFreeze(type));
+  }
+
+  public getType(type: string) {
+    return this.types.get(type);
+  }
+
+  public getAllTypes() {
+    return [...this.types.values()];
+  }
+
+  public isNamespaceAgnostic(type: string) {
+    return this.types.get(type)?.namespaceAgnostic ?? false;
+  }
+
+  public isHidden(type: string) {
+    return this.types.get(type)?.hidden ?? false;
+  }
+
+  public getIndex(type: string) {
+    return this.types.get(type)?.indexPattern;
+  }
```

Comment:
Yea, this is a tricky one. legacy saved objects were not forced to register a schema for a type, only it's mapping. Meaning that everything schema-related can be null, so this test was needed / made almost sense.

Now, we populate the type with default values when registered from legacy, meaning that a type not known by the registry is really not registered. However a lot of legacy/FTR tests assume that this can be called on unregistered types, that's why throwing seems like a little aggressive.

Reply:
Reasonable to me, thanks for explaining

Reply:
Will add proper JSdoc on the registry methods